ARMmbed/mbedtls
# fuzzing: better corpus for client and server

## Description
This follows up #1622 by having corpus for client and server that manage past `mbedtls_ssl_handshake`

To do this, the modifications were :
- removing `dummy_random` as was done in #2681 
- changing entropy initialization to use same key `ssl_client2`
- move up `srand` to make sure it is called before entropy is needed
- For the server, turn on tickets by default as is done with ssl_server2
- For the server, I modified ssl_server2 to use `mbedtls_test_srv_crt` as the fuzz target (instead of `mbedtls_test_srv_ec_crt`)

Plus the corpus files are changed and the documentation on how to generate it got better

## Status
**READY**

## Requires Backporting
 NO  

## Migrations
NO

## Todos

I do not think there is something else to do here.
We could add a test to check that ssl_client manages handshake on corpus file, but I do not see how we can manage this...

## Steps to test or reproduce
I compiled with cmake with `perl scripts/config.pl set MBEDTLS_PLATFORM_TIME_ALT` first in a build directory
Then from this build directory, I run `./programs/fuzz/fuzz_server ../programs/fuzz/corpuses/server` and check that `mbedtls_ssl_handshake` returns 0 with debug printing...
